pkg/utils/test/matchers/jq/jq_support_test.go

```diff
@@ -4,50 +4,168 @@ package jq
 import (
 	"encoding/json"
 	"reflect"
+	"strings"
 	"testing"
 
 	"github.com/onsi/gomega/gbytes"
 
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+
 	. "github.com/onsi/gomega"
 )
 
+func TestByteToType(t *testing.T) {
+	t.Parallel()
+
+	g := NewGomegaWithT(t)
+
+	tests := []struct {
+		name     string
+		input    []byte
+		expected any
+	}{
+		{
+			name:     "Empty Input",
+			input:    []byte{},
+			expected: nil,
+		},
+		{
+			name:     "Valid JSON Object",
+			input:    []byte(`{"key": "value"}`),
+			expected: map[string]any{"key": "value"},
+		},
+		{
+			name:     "Valid JSON Array",
+			input:    []byte(`[1, "two", 3.0]`),
+			expected: []any{float64(1), "two", 3.0},
+		},
+		{
+			name:     "Invalid JSON",
+			input:    []byte(`{invalid}`),
+			expected: nil,
+		},
+		{
+			name:     "Non-Object/Array JSON",
+			input:    []byte(`"string"`),
+			expected: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			result, err := bytesToType(tt.input)
+
+			if tt.expected == nil {
+				g.Expect(err).To(HaveOccurred())
+			} else {
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(result).To(Equal(tt.expected))
+			}
+		})
+	}
+}
+
 func TestToType(t *testing.T) {
 	t.Parallel()
 
 	typeTestData := []byte(`{ "foo": "bar" }`)
+
 	g := NewWithT(t)
 
-	items := map[string]func() any{
-		"gbytes": func() any {
-			b := gbytes.NewBuffer()
+	tests := []struct {
```

Comment:
It is also refactoring, not only "Add additional unit tests".

Reply:
I guess I should rename the commit ? :)

Reply:
> I guess I should rename the commit ? :)

Up to your feeling of beauty :) I'll not insist

Reply:
done